Marlin/src/lcd/menu/menu_probe_offset.cpp

```diff
@@ -107,47 +90,122 @@ void probe_offset_wizard_menu() {
 
   ACTION_ITEM(MSG_BUTTON_DONE, []{
     set_offset_and_go_back(calculated_z_offset);
-    do_z_clearance(20.0
-      #ifdef Z_AFTER_HOMING
-        - 20.0 + Z_AFTER_HOMING
-      #endif
+
+    // Set Z to 0, as we can expect
+    current_position.z = 0;
+    sync_plan_position();
+
+    // Raise Z as if it was homed
+    do_z_clearance(0
```

Comment:
```
do_z_clearance(20.0	
#ifdef Z_AFTER_HOMING
   - 20.0 + Z_AFTER_HOMING
#elseif defined(Z_HOMING_HEIGHT)
   + Z_HOMING_HEIGHT
#endif
);
```

Reply:
Of course it applies to the cancel part too.

Reply:
`#elif`

Reply:
@thinkyhead You are right.